### ISSUE TITLE
Position in mempool inaccurate on decimal

### ISSUE THREAD

**Issue by ziggamon** [spesmilo/electrum#4551]
Have noticed that when using sub-integer satoshi accuracy the "position in mempool" always places the transaction at the bottom of the interval, where in reality it's often on the top. 

How to reproduce (example):
See that there's a big block of transactions at 1.07 sat/B
https://www.buybitcoinworldwide.com/fee-calculator/
Make a transaction at 1.1 sat/B. It should be estimated as processed before those, but in practice it isn't and the position in mempool shows that there's 40 MB of tx ahead of it.

**Comment by SomberNight** [comment]
The client does not have the full mempool, only a "compact fee histogram". Currently, the highest possible precision in the fee histogram is whole sat/vbytes.

https://github.com/kyuupichan/electrumx/blob/73a46df17f4f2f57ef0ee5a9b0f66f6b5453eb14/electrumx/server/mempool.py#L376-L396

For example, at the moment the histogram one server is sending looks like this:
```
[Network] fee_histogram [[20, 105501], [10, 105204], [6, 178927], [5, 186548], [3, 80214], [1, 38040493]]
```
So the issue you mention is not easy to fix.

However, there is another issue:
that big chunk of txns around 1.07 sat/B is being put into the 1 sat/B bucket, and the client is suggesting to pay 1 sat/B when targeting e.g. 1 MB depth.

**Comment by SomberNight** [comment]
Yes, indeed, the max precision of the fee histogram could be increased.
It's not as simple as changing from ints to decimals, as on the server, the current implementation is storing the cumulative transaction size for every integer sat/byte value and sums those up to create the buckets.
